### PR TITLE
Updated Algorithm PowderReduceP2D

### PR DESCRIPTION
**Pull request by YannickMeinerzhagen**
**Description of work.**

The workflow algorithm PowderReduceP2D was using the deprecated algorithm AlignDetectors. The algorithm has been replaced by ApplyDiffCal and ConvertUnits as recommended in the documentation. Refinement of several datasets lead to equal results. 
Further the graphics for the workflow algorithm were updated to show these changes.
During testing an additional problem of the private function ResetNegatives2D was detected and fixed. For resetting negative Values to a specified value, a new workspace was created. The new workspace was not marked as a distribution (all other workspaces are distributions) and the x-Axis did not have a label. Due to these problems, calculations between workspaces did not work. These problems are now solved as well.
In order to use the algorithm with u to date measuring data from PG3 at SNS, an option for a background measurement specific to the vanadium measurement was included. It is now possible to reduce present measuring data from PG3. 

I am not sure if release notes are necessary. Maybe you have some advise for me?

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by YannickMeinerzhagen** [comment]
I have no idea, why the tests are failing... Some help would be much appreciated :)

**Comment by YannickMeinerzhagen** [comment]
Using the old code for powderreducep2d does not reproduce the error => the error got introduced by the changes made

commenting out the line for setting the distribution still yields the error

**Comment by sf1919** [comment]
This probably will need a release note as users will see a change. There is some guidance about how to do release notes at https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html . The guidance is new and it doesn't have a section about when to add release notes so I will make a note to add that.

**Comment by YannickMeinerzhagen** [comment]
Okay, I will start working on the release notes in a bit :) thanks for your help in that matter :)

**Comment by YannickMeinerzhagen** [comment]
The error was not reproduced after commenting out all lines associated with applydiffcal. Next step is to check if only commenting out the load of applydiffcal parameters is enough to make the algorithm run smoothly.

**Comment by YannickMeinerzhagen** [comment]
All errors related to the algorithm powderreducep2d are solved now. I have no idea about the errors that are shown in the failed tests, but they seem to be unrelated to the algorithm.

File: Framework/PythonInterface/plugins/algorithms/PowderReduceP2D.py [hunk 1]
```diff
@@ -652,12 +659,17 @@ def ResetNegatives2D(self, wsName, addMin, resetValue):
                             DataY=yDataNew,
                             DataE=eData,
                             NSpec=mtd[wsName].getNumberHistograms(),
+                            UnitX='dSpacing',
                             ParentWorkspace=mtd[wsName])
+            mtd[wsName].setDistribution(True)
```
Comment (RichardWaiteSTFC) [inline Framework/PythonInterface/plugins/algorithms/PowderReduceP2D.py:664]: It's a small thing but `ADS.retrieve(wsName)` is preferred to mtd . You can import it like so
`from mantid.api import AnalysisDataService as ADS`

File: Framework/PythonInterface/plugins/algorithms/PowderReduceP2D.py [hunk 2]
```diff
@@ -222,13 +227,12 @@ def loadMaskDetectors():
...
             self.setPropertyGroup('ComponentList', grp7)
 
-        def loadAlignDetectors():
-            # Input for AlignDetectors
-            #self.copyProperties('AlignDetectors', ['CalibrationWorkspace', 'OffsetsWorkspace'])
-            #grp8 = 'AlignDetectors'
-            #self.setPropertyGroup('CalibrationWorkspace', grp8)
-            #self.setPropertyGroup('OffsetsWorkspace', grp8)
-            pass
+        #def loadApplyDiffCal():
```
Comment (RichardWaiteSTFC) [inline Framework/PythonInterface/plugins/algorithms/PowderReduceP2D.py:230]: Was this commented block intended to be removed?

File: Framework/PythonInterface/plugins/algorithms/PowderReduceP2D.py [hunk 3]
```diff
@@ -393,7 +397,7 @@ def loadResetNegatives2DVana():
         loadRemovePromptPulse()
         loadLoadDiffCal()
         loadMaskDetectors()
-        loadAlignDetectors()
+        #loadApplyDiffCal ()
```
Comment (RichardWaiteSTFC) [inline Framework/PythonInterface/plugins/algorithms/PowderReduceP2D.py:400]: Is this meant to be commented out?

**Review by RichardWaiteSTFC — COMMENTED** [review]
Just a few small comments - I can't test this functionally without instructions, so will leave someone from SNS to pick this up and do a thorough review!

**Comment by YannickMeinerzhagen** [comment]
Thanks a lot for your comments, I have implemented them :)

**Comment by sf1919** [comment]
Work is now finished on v6.4 of Mantid and it will be released soon. As release editor I'm checking PRs before I go on leave to ensure any with v6.4 release notes move them to v6.5. I notice this PR is one of those - please can you move the release notes?

**Pull request closed without merging**